pyproject.toml

```diff
@@ -35,6 +35,7 @@ dependencies = [
   "networkx",
   "numpy",
   "pandas",
+  "scikit-learn",
```

Comment:
Make optional dependency? This can cause problems in conda environments with some scikit-learn versions requiring old or new NumPy versions which can cause problems with other packages that do not support NumPy 2.0 or require it.